test/integration_test.js

```diff
@@ -620,21 +623,48 @@ describe("Integration tests", () => {
           });
 
           describe(".createRecord()", () => {
-            it("should create a record", () => {
-              return coll
-                .createRecord({title: "foo"})
-                .should.eventually.have.property("data")
-                    .to.have.property("title").eql("foo");
+            describe("No record id provided", () => {
+              it("should create a record", () => {
+                return coll
+                  .createRecord({title: "foo"})
+                  .should.eventually.have.property("data")
+                      .to.have.property("title").eql("foo");
+              });
+
+              describe("Safe option", () => {
+                it("should check for existing record", () => {
+                  return coll.createRecord({title: "foo"})
+                    .then(({data}) => coll.createRecord({
+                      id: data.id,
+                      title: "foo"
+                    }, {safe: true}))
+                    .should.be.rejectedWith(Error, /412 Precondition Failed/);
+                });
+              });
             });
 
-            describe("Safe option", () => {
-              it("should check for existing record", () => {
-                return coll.createRecord({title: "foo"})
-                  .then(({data}) => coll.createRecord({
-                    id: data.id,
-                    title: "foo"
-                  }, {safe: true}))
-                  .should.be.rejectedWith(Error, /412 Precondition Failed/);
+            describe("Record id provided", () => {
+              const record = {
+                id: "37f727ed-c8c4-461b-80ac-de874992165c",
+                title: "foo"
+              };
+
+              it("should create a record", () => {
+                return coll
+                  .createRecord(record)
+                  .should.eventually.have.property("data")
+                      .to.have.property("title").eql("foo");
+              });
+
+              describe("Safe option", () => {
+                it("should check for existing record", () => {
+                  return coll.createRecord(record)
+                    .then(({data}) => coll.createRecord({
+                      id: data.id,
+                      title: "foo"
+                    }, {safe: true}))
+                    .should.be.rejectedWith(Error, /412 Precondition Failed/);
```

Comment:
IMO this test is redundant with the other one, since you test that the second call is rejected.